DNN Platform/Library/Common/Utilities/ValidationUtils.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DotNetNuke.Entities.Host;
+using DotNetNuke.Security;
+
+namespace DotNetNuke.Common.Utilities
+{
+    public sealed class ValidationUtils
+    {
+        internal static string GetDecryptionKey()
+        {
+            var machineKey = Config.GetDecryptionkey();
+            var key = $"{machineKey ?? ""}{Host.GUID.Replace("-", string.Empty)}";
+            return FIPSCompliant.EncryptAES(key, key, Host.GUID);
+        }
+
+        internal static string ComputeValidationCode(IList<string> extensions)
```

Comment:
Can we add something more dynamic to this validation code?  Maybe a module ID?  Otherwise, if there are two uploads (e.g. journal upload and profile picture upload), I could copy the validation code from one to the other and bypass the restriction that way.

Reply:
I guess even just a random salt that we sent to the client alongside the validation code would work

Reply:
After discussion, it sounds like these sorts of changes make sense in this context.

Reply:
After further discussion, we've talked about adding portal ID (or GUID?) and user ID, which would provide some benefit.  The remaining issue would be around expiration the validation code, but we can accept this before that's handled.

Reply:
Thanks @bdukes , added the portal id and user id in the check.